pkg/supportbundle/collect.go

```diff
@@ -73,70 +74,74 @@ func runCollectors(collectors []*troubleshootv1beta2.Collect, additionalRedactor
 	collectSpecs = collect.EnsureCollectorInList(collectSpecs, troubleshootv1beta2.Collect{ClusterResources: &troubleshootv1beta2.ClusterResources{}})
 	collectSpecs = collect.EnsureClusterResourcesFirst(collectSpecs)
 
-	var allCollectors []collect.Collector
-
 	allCollectedData := make(map[string][]byte)
 
 	k8sClient, err := kubernetes.NewForConfig(opts.KubernetesRestConfig)
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to instantiate Kubernetes client")
 	}
 
+	newDesiredCollectors := make(map[reflect.Type][]collect.Collector)
+
 	for _, desiredCollector := range collectSpecs {
 		if collectorInterface, ok := collect.GetCollector(desiredCollector, bundlePath, opts.Namespace, opts.KubernetesRestConfig, k8sClient, opts.SinceTime); ok {
 			if collector, ok := collectorInterface.(collect.Collector); ok {
 				err := collector.CheckRBAC(context.Background(), collector, desiredCollector, opts.KubernetesRestConfig, opts.Namespace)
 				if err != nil {
 					return nil, errors.Wrap(err, "failed to check RBAC for collectors")
 				}
-
-				if mergeCollector, ok := collectorInterface.(collect.MergeableCollector); ok {
-					allCollectors, err = mergeCollector.Merge(allCollectors)
-					if err != nil {
-						msg := fmt.Sprintf("failed to merge collector: %s: %s", collector.Title(), err)
-						opts.CollectorProgressCallback(opts.ProgressChan, msg)
-					}
-				} else {
-					allCollectors = append(allCollectors, collector)
-				}
+				collectorType := reflect.TypeOf(collector)
+				newDesiredCollectors[collectorType] = append(newDesiredCollectors[collectorType], collector)
 			}
 		}
 	}
 
-	foundForbidden := false
-	for _, c := range allCollectors {
-		for _, e := range c.GetRBACErrors() {
-			foundForbidden = true
-			opts.ProgressChan <- e
+	var foundForbidden bool
+	for collectorType, collectors := range newDesiredCollectors {
+		if mergeCollector, ok := collectors[0].(collect.MergeableCollector); ok {
```

Comment:
not in love with this, but i'm not sure of a better way to accomplish what I want. I would rather that the key in the `newDesiredCollectors` map could have a value that I could type check on, but I wasn't able to find a good way to do that.